src/api/app.py

```diff
@@ -167,26 +167,56 @@ def stream_chat(
             )
         )
 
-    try:
-        state_updates["messages"] = messages
-        state_updates["user_persona"] = user_persona
+    state_updates["messages"] = messages
+    state_updates["user_persona"] = user_persona
 
+    try:
         stream = zeno.stream(
             state_updates,
             config=config,
             stream_mode="updates",
             subgraphs=False,
         )
+        
         for update in stream:
-            node = next(iter(update.keys()))
-            yield pack(
-                {
-                    "node": node,
-                    "update": dumps(update[node]),
-                }
-            )
+            try:
+                node = next(iter(update.keys()))
+                yield pack(
+                    {
+                        "node": node,
+                        "update": dumps(update[node]),
+                    }
+                )
+            except Exception as e:
+                # Send error as a stream event instead of raising
+                yield pack(
+                    {
+                        "node": "error",
+                        "update": dumps({
+                            "error": True,
+                            "message": str(e),  # String representation of the error
```

Comment:
This does not seem like a good approach for production, we should log the error instead and only return a simple error message instead of an error dump.

Reply:
`str(e)` will print just the error message & not the entire stack trace.